src/components/RegistryAdmin.tsx

```diff
@@ -1,110 +1,163 @@
-  // refresh the accessToken.
-  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const [libraries, setLibraries] = useState<LibraryData[]>([]);
 
   const { accessToken, setAccessToken } = useContext(
     TokenContext
   ) as TokenContextValues;
 
-  // When a user clicks the "Log Out" button, the accessToken (in tokenContext)
-  // is reset to an empty string and the refreshToken is deleted from cookie
-  // storage.
+  // The logout function resets the accessToken to an empty string and deletes
+  // the refreshToken from cookie storage. This causes the LoginForm to be
+  // displayed.
   const logout = () => {
     setAccessToken('');
     Cookies.remove('refreshToken');
+    setIsLoading(false);
   };
 
   const refresh = () => {
     // Look for the refreshToken cookie.
     const refreshToken = Cookies.get('refreshToken');
 
-    // If a refreshToken exists, send it to the refresh endpoint.
+    // If a refreshToken exists, send it as a header to the refresh endpoint.
     if (refreshToken) {
-      fetch(process.env.QA_REFRESH as RequestInfo | URL, {
+      fetch(process.env.REFRESH as RequestInfo | URL, {
         method: 'POST',
         headers: { Authorization: `Bearer ${refreshToken}` },
       })
         .then((response) => {
           if (response.ok) {
             return response.json();
           } else {
-            // If the server responds with an error, setIsLoading to false.
-            // This will render the LoginForm.
-            setIsLoading(false);
             throw new Error('There was a problem refreshing.');
           }
         })
-        // If the server responds with a new accessToken, save it to
-        // tokenContext. Then, setIsLoading to false. This will render the
-        // LibrariesList.
+        // If the call is successful, save the new accessToken to tokenContext.
         .then((data) => {
           setAccessToken(data.access_token);
           setIsLoading(false);
         })
-        .catch((err) => console.log(err.message));
+        // If there is a problem refreshing, log the error message and use the
+        // logout function to delete the accessToken (if one exists) and the
+        // refreshToken. This will bring the user back to the LoginForm.
+        .catch((err) => {
+          console.log(err.message);
+          logout();
+        });
     } else {
-      // If there is no refreshToken, setIsLoading to false.
-      // This will render the LoginForm.
-      setIsLoading(false);
+      // If no refreshToken exists, delete the accessToken (if there is one)
+      // so the user is brought back to the LoginForm.
+      logout();
     }
   };
 
-  // If there is no accessToken in tokenContext, try refreshing. Else,
-  // setIsLoading to false. This will render the LibrariesList.
+  // Fetch the libraries with the accessToken.
+  const fetchLibraries = () => {
+    fetch(process.env.FETCH_LIBRARIES as RequestInfo | URL, {
```

Comment:
I think there is an opportunity here to refactor some fetch requests into a utility (something like `authenticatedFetch` or `fetchWithHeaders`) that manages all of the refresh logic, since this logic will be the same for changing a library's status. I think we should do this in a separate PR though, perhaps the next one where we add the library status fetch requests. Just wanted to call it out here.

Reply:
Sounds like a good idea. Okay!